### PR TITLE
*Added "delete account function"

### PR DESCRIPTION
**Pull request by maxonchil**
*Remove JWT from LS, when user delete account and redirect him to
home page
*Separated http request handlers from ther router and move them to
'handlers' folder
*Make an error handler like  separate module
*Added logg's at every succsesful action
*Some little improves
*Added writeLog middlewar for every incoming request
*Please dont kill for angular code

### CODE REVIEW

File: homework3/semi-uber/Server/api/handlers/user.handlers/userDeteleHandler.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const log4js = require("log4js");
+const logger = log4js.getLogger();
+const { User } = require("../../Schemas/user.schema");
+const errorHandler = require("../error.handler");
+
+async function userDeleteHandler(req, res) {
```
Comment (MaxHouz) [inline homework3/semi-uber/Server/api/handlers/user.handlers/userDeteleHandler.js]: You do not need to write `async`, if you are not using `await` in the function

File: homework3/semi-uber/Server/api/handlers/user.handlers/userDeteleHandler.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    .then(() => {
+      logger.info("User's account was successfully deleted");
+      res.json({
+        success: true,
+        data: {},
+        message: "User's account was successfully deleted"
+      });
+    })
+    .catch(error => {
+      return errorHandler(error.message, res);
```
Comment (MaxHouz) [inline homework3/semi-uber/Server/api/handlers/user.handlers/userDeteleHandler.js]: That's a nice approach, I like it

File: homework3/semi-uber/Server/api/handlers/user.handlers/userPatchHandler.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+const saltRounds = Number(config.get("saltRounds"));
+const errorHandler = require("../error.handler");
+
+const hashPassword = (password, saltRounds) => {
+  return bcrypt.hash(password, saltRounds);
+};
+
+async function userPatchHandler(req, res) {
+  const { newPassword, id: userID } = req.body;
+  const hashedPass = await hashPassword(newPassword, saltRounds);
```
Comment (MaxHouz) [inline homework3/semi-uber/Server/api/handlers/user.handlers/userPatchHandler.js]: If something goes wrong with password hashing, will it just return some falsy value, or an error might happen? Please check this scenario and make sure you will not get some unexpected errors here

File: homework3/semi-uber/Server/api/handlers/user.handlers/userPatchHandler.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  if (!hashPassword) {
+    return errorHandler("Password chanching was failt!", res);
+  }
+  const updatedUser = await User.findByIdAndUpdate(userID, {
```
Comment (MaxHouz) [inline homework3/semi-uber/Server/api/handlers/user.handlers/userPatchHandler.js]: Same here, I guess you need try/catch for such code. `if (!updatedUser)` will not catch an error, after something goes wrong on this line, the code execution will no go further.

File: homework3/semi-uber/Server/api/routes/login.router.js [hunk 5]
```diff
@@ -5,33 +5,29 @@ const { secret } = config.get("JWT");
...
-    data: {},
-    error: { code: 400, message }
-  });
-};
-
-router.post("/", async (req, res) => {
+router.post("/", writeLog, async (req, res) => {
   const { password, username } = req.body;
 
   const user = await User.findOne({ username });
```
Comment (MaxHouz) [inline homework3/semi-uber/Server/api/routes/login.router.js]: same regarding errors

File: homework3/semi-uber/Server/api/routes/login.router.js [hunk 6]
```diff
@@ -5,33 +5,29 @@ const { secret } = config.get("JWT");
...
   if (!user) {
-    return errorHandle("Login failed", res);
+    return errorHandler("Login failed", res);
   }
   const token = await jwt.sign(user.id, secret);
   if (!token) {
-    return errorHandle("Login failed", res);
+    return errorHandler("Login failed", res);
   }
   const verify = await bcrypt.compare(password, user.password);
```
Comment (MaxHouz) [inline homework3/semi-uber/Server/api/routes/login.router.js]: same regarding errors

File: homework3/semi-uber/Server/api/routes/login.router.js [hunk 7]
```diff
@@ -5,33 +5,29 @@ const { secret } = config.get("JWT");
...
-router.post("/", async (req, res) => {
+router.post("/", writeLog, async (req, res) => {
   const { password, username } = req.body;
 
   const user = await User.findOne({ username });
   if (!user) {
-    return errorHandle("Login failed", res);
+    return errorHandler("Login failed", res);
   }
   const token = await jwt.sign(user.id, secret);
```
Comment (MaxHouz) [inline homework3/semi-uber/Server/api/routes/login.router.js]: same regarding errors

File: homework3/semi-uber/Server/api/handlers/user.handlers/userDeteleHandler.js (hunk 1 above)
Comment (maxonchil) [reply homework3/semi-uber/Server/api/handlers/user.handlers/userDeteleHandler.js]: done

File: homework3/semi-uber/Server/api/handlers/user.handlers/userDeteleHandler.js (hunk 2 above)
Comment (maxonchil) [reply homework3/semi-uber/Server/api/handlers/user.handlers/userDeteleHandler.js]: Thanks<3

File: homework3/semi-uber/Server/api/handlers/user.handlers/userPatchHandler.js (hunk 4 above)
Comment (maxonchil) [reply homework3/semi-uber/Server/api/handlers/user.handlers/userPatchHandler.js]: Fixed

File: homework3/semi-uber/Server/api/handlers/user.handlers/userPatchHandler.js (hunk 3 above)
Comment (maxonchil) [reply homework3/semi-uber/Server/api/handlers/user.handlers/userPatchHandler.js]: Fixed

File: homework3/semi-uber/Server/api/routes/login.router.js (hunk 6 above)
Comment (maxonchil) [reply homework3/semi-uber/Server/api/routes/login.router.js]: Fixed

File: homework3/semi-uber/Server/api/routes/login.router.js (hunk 5 above)
Comment (maxonchil) [reply homework3/semi-uber/Server/api/routes/login.router.js]: Fixed

File: homework3/semi-uber/Server/api/routes/login.router.js (hunk 7 above)
Comment (maxonchil) [reply homework3/semi-uber/Server/api/routes/login.router.js]: Fixed

**Pull request merged**